src/test/regress/expected/pg12.out

```diff
@@ -361,8 +361,61 @@ SELECT DISTINCT y FROM test;
  25
 (1 row)
 
+-- non deterministic collations
+select public.run_command_on_master_and_workers($$
+CREATE COLLATION test_pg12.case_insensitive (
+	provider = icu,
+	locale = 'und-u-ks-level2',
+	deterministic = false
+);
+$$);
+ run_command_on_master_and_workers 
+-----------------------------------
+ 
+(1 row)
+
+select worker_hash('asdf'::text) asdf, worker_hash('aSdf'::text) "aSdf",
+	worker_hash('asdf'::text collate case_insensitive) "Casdf",
+	worker_hash('aSdf'::text collate case_insensitive) "CaSdf";
```

Comment:
Ah, I guess it also affects hash distribution

Reply:
One thing is that this test is using a non-deterministic collation. For deterministic collations, it should not impact hash distribution. Not sure how best to implement discerning the two, so current logic will not recognize different deterministic collations as colocated

Reply:
Issue: this test now fails because ICU changes hash values between versions (Ctrl+F for 'Sort keys will change from one ICU version to another' on [docs](http://userguide.icu-project.org/collation/api))

This only matters for non deterministic collations for hashing. Postgres stores version info for collations to warn user when collation version doesn't match their ICU version: https://www.postgresql.org/docs/10/sql-altercollation.html#SQL-ALTERCOLLATION-NOTES

Ideally we'd have a UDF to go through all records & move them to their appropriate shard. Would be useful also for use cases like changing ranges of range partition tables. Not sure how much this overlaps with shard rebalancer

Reply:
> I think the best option is to reject hash distribution on columns with non deterministic collations. I expect realistic use cases would want range distribution anyways

I think that's acceptable.